src/shared/buttercup/entries.js

```diff
@@ -50,77 +51,132 @@ export function validateEntry(entry) {
     }
   }
 
-  return errorMessages.length === 0
-    ? Promise.resolve(filterEmptyEntryValues(entry))
-    : Promise.reject(errorMessages.join('\n'));
+  if (errorMessages.length > 0) {
+    throw new Error(errorMessages.join('\n'));
+  }
+
+  return filterEmptyEntryValues(entry);
 }
 
-export function loadEntries(archiveId, groupId) {
+export async function loadEntries(archiveId, groupId) {
```

Comment:
how fast is this function going to be? Can't we load the icons async?

Reply:
Yes yes. Missed this one. It should probably behave in the same way as the `createEntry` and `updateEntry`. First load the entry data fast without waiting for the icons, then fetch the icons for each entry (either disk or downloaded) and dispatch updates

Reply:
Or, should it actually wait **only** for those icons which are already stored in disk, and defer only the real slow part, downloading + storing + reading from disk icons for those entries which did not have a stored icon? (This 2nd part I actually missed to implement for `loadEntries`)

Not sure if there would really be a big noticeable difference for those entries with icons stored in disk (the usual case). If waiting, it would be a bit slower, but it would avoid tons of dispatch(updateEntry) for each entry in the loaded list.